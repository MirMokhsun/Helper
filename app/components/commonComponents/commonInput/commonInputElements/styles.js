import { StyleSheet } from 'react-native';

export const styles = StyleSheet.create({
    container: {
        justifyContent: "flex-start",
        height: 15,
    },
    warningInput: {
        color: "#F86F6F",
        fontFamily: 'Roboto-Light',
        fontSize: 11,
        lineHeight: 15,
        textAlign:'right',
    }
});