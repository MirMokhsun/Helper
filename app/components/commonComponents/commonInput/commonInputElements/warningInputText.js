/* eslint-disable react/destructuring-assignment */
import React from "react";
import { View, Text } from 'react-native';
import { styles } from './styles';

export class WarningInputText extends React.PureComponent {
    constructor(props) {
        super(props)
        this.state = {
            warningInputText: null
        }
    }

    UNSAFE_componentWillReceiveProps({ isShowWarningText, text, testID, maxWidth }) {
        let result = (<View style={{ height: 15 }} />);
        if (isShowWarningText) {
            result =
                (<View style={[styles.container, { width: maxWidth || 290 }]}>
                    <Text
                        style={styles.warningInput}
                        testID={testID}
                        accessibilityLabel={testID}>
                        {text}
                    </Text>
                </View>);
        }
        this.setState({ warningInputText: result });
    }

    render() {
        return this.state.warningInputText;
    }
}