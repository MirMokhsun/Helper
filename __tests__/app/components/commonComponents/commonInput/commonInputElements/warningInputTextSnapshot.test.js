import 'react-native';
import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import { WarningInputText } from '../../../../../../app/components/commonComponents/commonInput/commonInputElements/warningInputText';
import renderer from 'react-test-renderer';

describe('WarningInputText', () => {
    it('create snapshot WarningInputText', () => {
        const renderer = new ShallowRenderer();
        let warningInputText = renderer.render(<WarningInputText maxWidth={260} isShowWarningText={true} text={'errorText'} testID={'textErrorID'} />);
        expect(warningInputText).toMatchSnapshot();
    });
    describe('test componentWillReceiveProps', () => {
        it('test with next props isShowWarningText: true', () => {
            let warningInputText = renderer.create(<WarningInputText maxWidth={260} isShowWarningText={true} text={'errorText'} testID={'textErrorID'} />).getInstance();
            let nextProps = {
                isShowWarningText: true,
                text: 'some text',
            };
            expect(warningInputText.state.warningInputText).toBeNull();
            warningInputText.componentWillReceiveProps(nextProps);
            let warningInputTextState = renderer.create(warningInputText.state.warningInputText);
            expect(warningInputTextState).not.toBeNull();
            expect(warningInputTextState).toMatchSnapshot();
        });
        it('test with next props isShowWarningText: false', () => {
            let warningInputText = renderer.create(<WarningInputText maxWidth={260} isShowWarningText={true} text={'errorText'} testID={'textErrorID'} />).getInstance();
            let nextProps = {
                isShowWarningText: false,
                text: 'some text',
            };
            expect(warningInputText.state.warningInputText).toBeNull();
            warningInputText.componentWillReceiveProps(nextProps);
            let warningInputTextState = renderer.create(warningInputText.state.warningInputText);
            expect(warningInputTextState).not.toBeNull();
            expect(warningInputTextState).toMatchSnapshot();

        });
    });
});