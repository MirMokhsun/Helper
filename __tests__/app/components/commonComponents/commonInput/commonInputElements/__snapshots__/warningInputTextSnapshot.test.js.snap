// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WarningInputText create snapshot WarningInputText 1`] = `null`;

exports[`WarningInputText test componentWillReceiveProps test with next props isShowWarningText: false 1`] = `
<View
  style={
    Object {
      "height": 15,
    }
  }
/>
`;

exports[`WarningInputText test componentWillReceiveProps test with next props isShowWarningText: true 1`] = `
<View
  style={
    Array [
      Object {
        "height": 15,
        "justifyContent": "flex-start",
      },
      Object {
        "width": 290,
      },
    ]
  }
>
  <Text
    style={
      Object {
        "color": "#F86F6F",
        "fontFamily": "Roboto-Light",
        "fontSize": 11,
        "lineHeight": 15,
        "textAlign": "right",
      }
    }
  >
    some text
  </Text>
</View>
`;
